crates/uv-distribution/src/metadata/build_requires.rs

```diff
@@ -205,14 +204,20 @@ impl BuildRequires {
     }
 }
 
-/// Lowered extra build dependencies with source resolution applied.
+/// Lowered extra build dependencies.
+///
+/// This is a wrapper around [`ExtraBuildRequires`] that provides methods to lower
+/// [`ExtraBuildDependencies`] from a workspace context or from already lowered dependencies.
 #[derive(Debug, Clone, Default)]
-pub struct ExtraBuildRequires {
-    pub extra_build_dependencies: ExtraBuildDependencies,
-}
+pub struct LoweredExtraBuildDependencies(ExtraBuildRequires);
```

Comment:
This is a wrapper type, mostly to avoid circular dependencies. We want `ExtraBuildRequires` to be usable in the build dispatch traits, but those can't depend on `uv-distribution` (and yet we want the lowering logic to live here). We also use this pattern for at least one other type.